ci/tests/Jenkinsfile.e2e-upgrade

```diff
@@ -0,0 +1,84 @@
+pipeline {
+
+  agent { label 'linux' }
+
+  parameters {
+    string(
+      name: 'NETWORK',
+      description: 'Name of test network to use.',
+      defaultValue: 'ropsten',
+    )
+    string(
+      name: 'APK_NAME',
+      description: 'Filename of APK uploaded to SauceLabs',
+    )
+     string(
+      name: 'APK_NAME_UPGRADE',
+      description: 'Filename of APK of upgraded application',
+    )
+  }
+
+  options {
+    disableConcurrentBuilds()
+  }
+
+
+  stages {
+    stage('Setup') {
+      steps { script {
+        dir('test/appium') {
+          sh 'pip3 install --user -r requirements.txt'
+        }
+      } }
+    }
+    stage('Test') {
+      steps {
+        withCredentials([
+          usernamePassword(
+            credentialsId:  'test-rail-api',
+            usernameVariable: 'TESTRAIL_USER',
+            passwordVariable: 'TESTRAIL_PASS'
+          ),
+          usernamePassword(
+            credentialsId:  'sauce-labs-api',
+            usernameVariable: 'SAUCE_USERNAME',
+            passwordVariable: 'SAUCE_ACCESS_KEY'
+          ),
+          string(
+            credentialsId: 'etherscan-api-key',
+            variable: 'ETHERSCAN_API_KEY'
+          ),
+        ]) {
+          dir('test/appium/tests') {
+            sh """
+              python3 -m pytest -m upgrade \
+                -n24 --rerun_count=2 \
+                --testrail_report=True \
+                --network=${params.NETWORK} \
+                --apk=${params.APK_NAME} \
+                --apk_upgrade=${params.APK_NAME_UPGRADE}
```

Comment:
Do we even need a separate Jenkinsfile? Couldn't we just make this `--apk_upgrade` argument optional if `APK_NAME_UPGRADE` is set"? Or does that change the type of tests run too much?

Reply:
no, we can't I guess. I see 2 reasons here:
1)we need to exclude testcases for "upgrade" for nightly builds (-k 'upgrade') and launch with -m `upgrade` for upgrade job
2) --apk=${params.APK_NAME}, which is used as main app in nightly job should be current release apk in case of upgrade job.

Reply:
Fair enough.